emcc.py

```diff
@@ -906,7 +903,7 @@ def need_llvm_debug_info():
       options.memory_init_file = shared.Settings.OPT_LEVEL >= 2
 
     # TODO: support source maps with js_transform
-    if options.js_transform and shared.Settings.GENERATE_SOURCE_MAP:
+    if options.js_transform and shared.Settings.DEBUG_LEVEL >= 4:
```

Comment:
This was never firing because GENERATE_SOURCE_MAP is never assigned until later on.